src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -1286,7 +1403,240 @@ static int _xccdf_policy_generate_fix_other(struct oscap_list *rules_to_fix, str
+	oscap_iterator_free(service_disable_it);
+	oscap_iterator_free(service_enable_it);
+	return 0;
+}
+
+static int _generate_kickstart_packages(struct kickstart_commands *cmds, int output_fd)
+{
+	_write_text_to_fd(output_fd, "# Packages selection (%packages section is required)\n");
+	_write_text_to_fd(output_fd, "%packages\n");
+	/* openscap-scanner and scap-security-guide needs to be installed because we will run oscap in the %post section */
+	_write_text_to_fd(output_fd, "openscap-scanner\n");
+	_write_text_to_fd(output_fd, "scap-security-guide\n");
+	struct oscap_iterator *package_install_it = oscap_iterator_new(cmds->package_install);
+	while (oscap_iterator_has_more(package_install_it)) {
+		char *package = (char *) oscap_iterator_next(package_install_it);
+		_write_text_to_fd(output_fd, package);
+		_write_text_to_fd(output_fd, "\n");
+	}
+	oscap_iterator_free(package_install_it);
+	struct oscap_iterator *package_remove_it = oscap_iterator_new(cmds->package_remove);
+	while (oscap_iterator_has_more(package_remove_it)) {
+		char *package = (char *) oscap_iterator_next(package_remove_it);
+		_write_text_to_fd(output_fd, "-");
+		_write_text_to_fd(output_fd, package);
+		_write_text_to_fd(output_fd, "\n");
+	}
+	oscap_iterator_free(package_remove_it);
+	_write_text_to_fd(output_fd, "%end\n");
+	_write_text_to_fd(output_fd, "\n");
+	return 0;
+}
+
+static int _generate_kickstart_post(struct kickstart_commands *cmds, const char *profile_id, const char *input_path, int output_fd)
+{
+	_write_text_to_fd(output_fd, "%post\n");
+	char *dup = strdup(input_path);
+	char *basename = oscap_basename(dup);
+	free(dup);
+	char *oscap_command = oscap_sprintf(
+		"oscap xccdf eval --remediate --profile '%s' /usr/share/xml/scap/ssg/content/%s\n",
+		profile_id, basename);
+	free(basename);
+	_write_text_to_fd(output_fd, "# Perform OpenSCAP hardening\n");
+	_write_text_to_fd_and_free(output_fd, oscap_command);
+	struct oscap_iterator *post_it = oscap_iterator_new(cmds->post);
+	while (oscap_iterator_has_more(post_it)) {
+		char *command = (char *) oscap_iterator_next(post_it);
+		_write_text_to_fd(output_fd, command);
+		_write_text_to_fd(output_fd, "\n");
+	}
+	oscap_iterator_free(post_it);
+	_write_text_to_fd(output_fd, "%end\n");
+	_write_text_to_fd(output_fd, "\n");
+	return 0;
+}
+
+
+const char *common_partition = (
+"# Initialize (format) all disks (optional)\n"
+"zerombr\n"
+"\n"
+"# The following partition layout scheme assumes disk of size 20GB or larger\n"
+"# Modify size of partitions appropriately to reflect actual machine's hardware\n"
+"#\n"
+"# Remove Linux partitions from the system prior to creating new ones (optional)\n"
+"# --all	erase all partitions\n"
+"# --initlabel	initialize the disk label to the default based on the underlying architecture\n"
+"clearpart --all --initlabel\n"
+"reqpart\n"
+"\n"
+"# Create primary system partitions (required for installs)\n"
+"part /boot --fstype=xfs --size=512 --fsoptions=\"nodev,nosuid,noexec\"\n"
```

Comment:
Isn't it already a part of `reqpart`?

Reply:
No, there is a difference between
```
reqpart
```
and
```
reqpart --add-boot
```
which should be pretty obvious. :smile:

Just `reqpart` adds stuff like
* 1 MB `biosboot` partition on GUID partition tables, for grub2 data
* 256 / 512 MB `efi` partition on UEFI systems
* `PReP` boot partition on ppc64 / ppc64le
* etc.

(Basically hardware-related boot dependencies.)

Reply:
Oh, yeah, we are still missing `--add-boot`, okay.